src/client.ts

```diff
@@ -245,6 +268,7 @@ export class Client {
    * @return {String}     Encoded query string to be used in a search URL.
    *
    */
+  // TODO: All validation should be done in Query
   public buildSearchQueryString(query?: string | Query, params?: DoofinderParameters): string {
```

Comment:
@resetreboot if you want to take a look at this, I think all this validation should be done inside `Query`.